internal/telemetry/default.go

```diff
@@ -81,64 +75,15 @@ func (t *defaultTelemetry) RecordOperation(op OperationInvocation) {
 	}, float64(op.DurationMicros))
 }
 
-type startSessionResponse struct {
-	SessionID string `json:"sessionId"`
-}
-
 func (t *defaultTelemetry) ensureSession(ctx context.Context) {
-	if t.sessionID != "" {
+	if t.sessionID != 0 {
 		return
 	}
 
 	t.getSession.Do(func() {
 		logger := logging.CurrentLogger(ctx)
-		errSetSession := func(res *http.Response, err error) {
-			var msg strings.Builder
-			msg.WriteString("Error creating session. Setting null UUID.")
-			if res != nil && res.StatusCode > 0 {
-				var responseText string
-				if responseBody, resErr := ioutil.ReadAll(res.Request.Body); resErr != nil {
-					responseText = "cannot read response"
-				} else {
-					responseText = string(responseBody)
-				}
-				msg.WriteString(fmt.Sprintf(" - HTTP %d: %q", res.StatusCode, responseText))
-			}
-
-			if err != nil {
-				msg.WriteString(fmt.Sprintf(": %v", err))
-			}
-
-			logger.Infof(msg.String())
-			t.sessionID = "00000000-0000-0000-0000-000000000000"
-		}
-
-		var buf bytes.Buffer
-		req := telemetryRequest{
-			Method: "start-session",
-		}
-		e := json.NewEncoder(&buf)
-		e.Encode(req)
-		res, err := t.client.Post(exo.TelemetryEndpoint, "application/json", &buf)
-		if err != nil {
-			errSetSession(res, err)
-			return
-		}
-		defer res.Body.Close()
-
-		if res.StatusCode != 200 {
-			errSetSession(res, nil)
-			return
-		}
-
-		typedRes := startSessionResponse{}
-		if err := json.NewDecoder(res.Body).Decode(&typedRes); err != nil {
-			errSetSession(res, err)
-			return
-		}
-
-		t.sessionID = typedRes.SessionID
-		logger.Infof("Started session: %q", t.sessionID)
+		t.sessionID = chrono.NowMillisecond(ctx)
```

Comment:
this can run in the cli right? what if somebody does `exo whatever & exo whatever else` is there a session id race?

Reply:
It should be fine. They will be considered one session if they execute in the same millisecond and two sessions otherwise. This is one of the quirks that I found with Amplitude.

Reply:
ok, so sessions are not durable or persistent across multiple invocations of the CLI for example?